File: sdks/node/src/health.ts
export interface HealthCheckOptions {
  ingestUrl?: string;
  timeoutMs?: number;
}

export async function checkIngestHealth(
  ingestUrl = 'https://ingest.amlexia.com',
  options: HealthCheckOptions = {},
): Promise<{ ok: boolean; status: number; latencyMs: number }> {
  const base = ingestUrl.replace(/\/$/, '');
  const timeoutMs = options.timeoutMs ?? 5000;
  const start = Date.now();
  const controller = new AbortController();
  const timer = setTimeout(() => controller.abort(), timeoutMs);
  try {
    const res = await fetch(`${base}/health`, { signal: controller.signal });
    return { ok: res.ok, status: res.status, latencyMs: Date.now() - start };
  } catch {
    return { ok: false, status: 0, latencyMs: Date.now() - start };
  } finally {
    clearTimeout(timer);
  }
}

File: sdks/go/health.go
package amlexia

import (
	"net/http"
	"time"
)

type HealthResult struct {
	OK        bool `json:"ok"`
	Status    int  `json:"status"`
	LatencyMs int  `json:"latencyMs"`
}

func CheckIngestHealth(ingestURL string, timeout time.Duration) HealthResult {
	if timeout == 0 {
		timeout = 5 * time.Second
	}
	base := ingestURL
	if base == "" {
		base = DefaultIngestURL
	}
	start := time.Now()
	client := &http.Client{Timeout: timeout}
	res, err := client.Get(base + "/health")
	if err != nil {
		return HealthResult{OK: false, Status: 0, LatencyMs: int(time.Since(start).Milliseconds())}
	}
	defer res.Body.Close()
	return HealthResult{
		OK:        res.StatusCode >= 200 && res.StatusCode < 300,
		Status:    res.StatusCode,
		LatencyMs: int(time.Since(start).Milliseconds()),
	}
}
